clientapp/app/components/shared/Measurements.tsx

```diff
@@ -1,22 +1,57 @@
-import React from 'react'
-interface IMeasurementsProps {
-  measurementRequest:string,
-  placeHolderExample:string,
+'use client'
+import React, { ChangeEvent, FormEvent } from 'react';
+import { useContext } from 'react';
+import { MeasurementContext,IMeasurements } from './context/MeasurementsData';
+import { sendMeasurementsToApi } from '@/app/Api/post_measurements';
 
-}
+export default function Measurements() {
+
+  const context = useContext(MeasurementContext);
+
+  if (!context) {
```

Comment:
nit: `if (context == null)`

`if (!context)` tests if context is false, not nullish.  if context was undefined or null, this if block wouldnt run.